api/src/main/java/org/pmiops/workbench/billing/FreeTierBillingBatchUpdateService.java

```diff
@@ -38,38 +39,60 @@ public class FreeTierBillingBatchUpdateService {
 
   private final BigQueryService bigQueryService;
 
+  private final GoogleProjectPerCostDao googleProjectPerCostDao;
+
+  private final WorkspaceDao workspaceDao;
+
   private static final int MIN_USERS_BATCH = 5;
   private static final int MAX_USERS_BATCH = 999;
   public static final Range<Integer> batchSizeRange =
       Range.closed(MIN_USERS_BATCH, MAX_USERS_BATCH);
 
   @Autowired
   public FreeTierBillingBatchUpdateService(
+      GoogleProjectPerCostDao googleProjectPerCostDao,
       UserDao userDao,
       FreeTierBillingService freeTierBillingService,
       Provider<WorkbenchConfig> workbenchConfigProvider,
+      WorkspaceDao workspaceDao,
       BigQueryService bigQueryService) {
+    this.googleProjectPerCostDao = googleProjectPerCostDao;
     this.userDao = userDao;
     this.freeTierBillingService = freeTierBillingService;
     this.workbenchConfigProvider = workbenchConfigProvider;
     this.bigQueryService = bigQueryService;
+    this.workspaceDao = workspaceDao;
   }
 
   /**
    * Takes in List of : user and all BQ cost of workspace for the user
    *
-   * @param userBQCostReq
+   * @param userIdList
    */
-  public void checkAndAlertFreeTierBillingUsage(List<UserBQCost> userBQCostReq) {
+  public void checkAndAlertFreeTierBillingUsage(List<Long> userIdList) {
     Map<String, Double> userWorkspaceBQCosts = new HashMap<String, Double>();
 
+    Set<String> googleProjectsForUserSet =
+        userIdList.stream()
+            .map(
+                (userId) -> {
+                  return workspaceDao.getGoogleProjectForUser(userId);
+                })
+            .collect(HashSet::new, HashSet::addAll, HashSet::addAll);
+
+    // Create Map of project and cost
+    googleProjectsForUserSet.forEach(
+        (usersGoogleProject) -> {
+          userWorkspaceBQCosts.put(
+              usersGoogleProject,
+              googleProjectPerCostDao.getCostByGoogleProject(usersGoogleProject));
```

Comment:
I am concerned about the performance of this db transaction if the number of google projects is large, do you mind testing on Test env to see if the task won't time out? 
The other option is to make one db call by sending the set of google projects to query for...

Reply:
> The other option is to make one db call by sending the set of google projects to query for...


In complete, but I do feel this query and the one above can be optimized. 
Two DB query seems not necessary. And since we already doing smaller batches, we might be able to query make on call